src/components/flame-graph/FlameGraph.js

```diff
@@ -90,6 +97,120 @@ class FlameGraph extends React.PureComponent<Props> {
     }
   };
 
+  /**
+   * Is the box for this call node wide enough to be selected?
+   */
+  _wideEnough = (callNodeIndex: IndexIntoCallNodeTable): boolean => {
+    const { flameGraphTiming, callNodeInfo: { callNodeTable } } = this.props;
+
+    const depth = callNodeTable.depth[callNodeIndex];
+    const row = flameGraphTiming[depth];
+    const columnIndex = row.callNode.indexOf(callNodeIndex);
+    return row.end[columnIndex] - row.start[columnIndex] > SELECTABLE_THRESHOLD;
+  };
+
+  /**
+   * Return next keyboard selectable callNodeIndex along one
+   * horizontal direction.
+   *
+   * `direction` should be either -1 (left) or 1 (right).
+   *
+   * Returns undefined if no selectable callNodeIndex can be found.
+   * This means we're already at the end, or the boxes of all
+   * candidate call nodes are too narrow to be selected.
+   */
+  _nextSelectableInRow = (
+    startingCallNodeIndex: IndexIntoCallNodeTable,
+    direction: 1 | -1
+  ): ?IndexIntoCallNodeTable => {
```

Comment:
nit: here I think it returns `IndexIntoCallNodeTable | void`; remember that `?IndexIntoCallNodeTable` includes `null` as well.